photon-lib/src/main/scala/com/linkedin/photon/ml/evaluation/MultiEvaluator.scala

```diff
@@ -64,6 +64,37 @@ abstract class MultiEvaluator(
       .mean()
   }
 
+  /**
+   * Compute an evaluation metric on a per-group basis and save the per group results.
+   *
+   * @param scoresAndLabelsAndWeights A [[RDD]] of pairs (uniqueId, (score, label, weight))
+   * @return A tuple of evaluation metric mean value and per-group evaluation
+   */
+  def evaluatePerGroup(
+      scoresAndLabelsAndWeights: RDD[(UniqueSampleId, (Double, Double, Double))]):
+    (Double, Option[RDD[(String, Double)]]) = {
+
+    // Create a local copy of the localEvaluator, so that the underlying object won't get shipped to the executor nodes
+    val localEvaluator = this.localEvaluator
+
+    // EvaluationSuite guarantees that all validation data is in scoresAndLabelsAndWeights RDD, and ids RDD is directly
+    // mapped from validation data. Thus, inner join should be guaranteed to not lose any data.
+    val groupedData = scoresAndLabelsAndWeights
+      .join(ids)
+      .map { case (_, (scoreLabelAndWeight, id)) => (id, scoreLabelAndWeight) }
+      .groupByKey()
+
+      val perGroupEvaluation = groupedData
+        .mapValues(scoreLabelAndWeights => localEvaluator.evaluate(scoreLabelAndWeights.toArray))
+        .filter(results => !(results._2.isInfinite || results._2.isNaN))
```

Comment:
Curious on what circumstances we will see `inf` or `NaN`

Reply:
When an entity (e.g. member, job) in the validation doesn't have a model trained, the output is NaN. I'm not sure when will the inf happen, but I still put the condition here as it was previously.